src/main/java/com/limechain/storage/KVRepository.java

```diff
@@ -44,7 +44,7 @@ public interface KVRepository<K, V> {
      * @param limit maximum keys to return
      * @return whether the delete operation was successful
      */
-    List<byte[]> findKeysByPrefix(String prefixSeek, int limit);
+    List<byte[]> findKeysByPrefix(K prefixSeek, int limit);
```

Comment:
Are we sure all generic `K`s implementing this interface would have a semantically valid notion of what a "prefix" is?
(Not saying it was previously better and I clearly see your use case, just that we could note this as a potential issue)

Reply:
K should implement Serializable and/or Iterable interfaces. This is a large refactor which has been discussed and needs to happen in a separate task.